app/utils/categories.test.ts

```diff
@@ -0,0 +1,663 @@
+                lastViewedAt: 2,
+                isUnread: false,
+            } as MyChannelModel,
+        }, {
+            ...data[1],
+            myChannel: {
+                id: data[1].myChannel.id,
+                lastViewedAt: 1,
+                isUnread: false,
+            } as MyChannelModel,
+        }];
+
+        result = filterAutoclosedDMs('direct_messages', 3, 'me', '', mockRecent1, [], {});
+        expect(result[0].channel.id).toEqual('dm1');
+
+        const mockRecent2: ChannelWithMyChannel[] = [{
+            ...data[0],
+            myChannel: {
+                id: data[0].myChannel.id,
+                lastViewedAt: 1,
+                isUnread: false,
+            } as MyChannelModel,
+        }, {
+            ...data[1],
+            myChannel: {
+                id: data[1].myChannel.id,
+                lastViewedAt: 2,
+                isUnread: false,
+            } as MyChannelModel,
+        }];
+
+        result = filterAutoclosedDMs('direct_messages', 3, 'me', '', mockRecent2, [], {});
+        expect(result[0].channel.id).toEqual('dm2');
+
+        const mockSameLastViewed: ChannelWithMyChannel[] = [{
+            ...data[0],
+            myChannel: {
+                id: data[0].myChannel.id,
+                lastViewedAt: 1,
+                isUnread: false,
+            } as MyChannelModel,
+        }, {
+            ...data[1],
+            myChannel: {
+                id: data[1].myChannel.id,
+                lastViewedAt: 1,
+                isUnread: false,
+            } as MyChannelModel,
+        }];
+
+        result = filterAutoclosedDMs('direct_messages', 3, 'me', '', mockSameLastViewed, [], {});
+        expect(result[0].channel.id).toEqual('dm1');
+    });
+
+    test('filterManuallyClosedDms', async () => {
+        const db = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+
+        const closedDMsPrefs = [{
+            id: 'pref1',
+            userId: 'me',
+            category: Preferences.CATEGORIES.DIRECT_CHANNEL_SHOW,
+            name: 'dm1',
+            value: 'true',
+        }, {
+            id: 'pref2',
+            userId: 'me',
+            category: Preferences.CATEGORIES.DIRECT_CHANNEL_SHOW,
+            name: 'dm2',
+            value: 'true',
+        }, {
+            id: 'pref3',
+            userId: 'me',
+            category: Preferences.CATEGORIES.GROUP_CHANNEL_SHOW,
+            name: 'gm1',
+            value: 'false',
+        }] as PreferenceModel[];
+
+        const {channels, myChannels} = buildChannelsData();
+        const data = await createChannelsDbRecords(db, channels, myChannels);
+
+        const notifyProps = myChannels.reduce<Record<string, Partial<ChannelNotifyProps>>>((obj, m) => {
+            obj[m.channel_id] = m.notify_props;
+            return obj;
+        }, {});
+
+        const result = filterManuallyClosedDms(data, notifyProps, closedDMsPrefs, 'me');
+        expect(result.length).toEqual(3);
```

Comment:
We seem to be only testing for the happy path here. Is that intended?

Reply:
What you mean?

Reply:
We only make one call to filterManuallyCloseDms in the test, and that call actually doesn't filter any channel (3 channels are passed on data, and three are returned).

Reply:
Oh this ine needs fixing for sure.. maybe I overlooked it.. will add in a separate PR